File: oaktrees/src/main/java/co/com/oaktrees/dto/NuevaPersona.java
package co.com.oaktrees.dto;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import java.util.HashSet;
import java.util.Set;

public class NuevaPersona {

    @NotBlank
    private String clave;
    @NotBlank
    @Email
    private String correo;
    @NotBlank
    private String nombre;
    @NotBlank
    private String telefono;

    private Set<String> roles = new HashSet<>();

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public String getClave() {
        return clave;
    }

    public void setClave(String clave) {
        this.clave = clave;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }
}

File: oakFront/src/app/models/nueva-persona.ts
export class NuevaPersona {
  telefono: string;
  nombre: string;
  correo: string;
  clave: string;
  rol: string

  constructor(nombre: string, telefono: string, correo: string, clave: string, rol: string) {
    this.clave = clave;
    this.correo = correo;
    this.nombre = nombre;
    this.telefono = telefono;
    this.rol = rol;
  }

}
